app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -741,4 +747,26 @@ public void onRowClicked(int inputPosition)
             startActivityForResult(i, Utils.MAIN_REQUEST);
         }
     }
+
+    @Override
+    protected void onPostResume() {
+        super.onPostResume();
+
+        settings = new Settings(this);
+        if(settings.getAppLockStatus()){  // App Lock On
+            new UnlockDialogBuilder(this)
+                    .onUnlocked(() -> {
+                        appIsLocked = false;
+                        updateLoyaltyCardList();
+                        Toast.makeText(this, "Unlocked", Toast.LENGTH_SHORT).show();
```

Comment:
Please don't use hardcoded text messages, use R.string values

Even better, don't use a toast, not necessary to tell the user they unlocked it as it's obvious (dialog disappears and data appears)